File: sdk/php/src/Resources/AffiliatesAdvanced.php
<?php

namespace HiTechCloud\SDK\Resources;

class AffiliatesAdvanced extends BaseResource
{
    /**
     * Get affiliate info
     */
    public function getInfo(int $clientId): array
    {
        return $this->http->get('/api/affiliates_adv/' . $clientId . '/info');
    }

    /**
     * Get stats
     */
    public function getStats(int $clientId): array
    {
        return $this->http->get('/api/affiliates_adv/' . $clientId . '/stats');
    }

    /**
     * Get commission plans
     */
    public function getCommissionPlans(int $clientId): array
    {
        return $this->http->get('/api/affiliates_adv/' . $clientId . '/commission-plans');
    }

    /**
     * Get vouchers
     */
    public function getVouchers(int $clientId): array
    {
        return $this->http->get('/api/affiliates_adv/' . $clientId . '/vouchers');
    }

    /**
     * Get commissions
     */
    public function getCommissions(int $clientId): array
    {
        return $this->http->get('/api/affiliates_adv/' . $clientId . '/commissions');
    }

    /**
     * Get referrals
     */
    public function getReferrals(int $clientId): array
    {
        return $this->http->get('/api/affiliates_adv/' . $clientId . '/referrals');
    }

    /**
     * Get payouts
     */
    public function getPayouts(int $clientId): array
    {
        return $this->http->get('/api/affiliates_adv/' . $clientId . '/payouts');
    }

    /**
     * Get campaigns
     */
    public function getCampaigns(int $clientId): array
    {
        return $this->http->get('/api/affiliates_adv/' . $clientId . '/campaigns');
    }

    /**
     * Get audit log
     */
    public function getAudit(int $clientId): array
    {
        return $this->http->get('/api/affiliates_adv/' . $clientId . '/audit');
    }

    /**
     * Activate affiliate
     */
    public function activate(int $clientId): array
    {
        return $this->http->post('/api/affiliates_adv/' . $clientId . '/activate');
    }

    /**
     * Set commission plan
     */
    public function setCommissionPlan(int $clientId, int $commissionId): array
    {
        return $this->http->post('/api/affiliates_adv/' . $clientId . '/commission-plan/' . $commissionId);
    }

    /**
     * Generate voucher
     */
    public function generateVoucher(int $clientId, int $planId): array
    {
        return $this->http->post('/api/affiliates_adv/' . $clientId . '/vouchers/' . $planId);
    }

    /**
     * Delete voucher
     */
    public function deleteVoucher(int $clientId, int $voucherId): array
    {
        return $this->http->delete('/api/affiliates_adv/' . $clientId . '/vouchers/' . $voucherId);
    }

    /**
     * Set landing page
     */
    public function setLandingPage(int $clientId, array $data): array
    {
        return $this->http->post('/api/affiliates_adv/' . $clientId . '/landing-page', $data);
    }

}

File: sdk/csharp/Resources/AffiliatesAdvanced.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace HiTechCloud.SDK.Resources;

public class AffiliatesAdvanced : BaseResource
{
    public AffiliatesAdvanced(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> InfoAsync(int clientId)
    {
        return await Http.GetAsync(string.Format("/api/affiliates_adv/{clientId}/info", clientId));
    }

    public async Task<Dictionary<string, object?>> StatsAsync(int clientId)
    {
        return await Http.GetAsync(string.Format("/api/affiliates_adv/{clientId}/stats", clientId));
    }

    public async Task<Dictionary<string, object?>> CommissionPlansAsync(int clientId)
    {
        return await Http.GetAsync(string.Format("/api/affiliates_adv/{clientId}/commission-plans", clientId));
    }

    public async Task<Dictionary<string, object?>> VouchersAsync(int clientId)
    {
        return await Http.GetAsync(string.Format("/api/affiliates_adv/{clientId}/vouchers", clientId));
    }

    public async Task<Dictionary<string, object?>> CommissionsAsync(int clientId)
    {
        return await Http.GetAsync(string.Format("/api/affiliates_adv/{clientId}/commissions", clientId));
    }

    public async Task<Dictionary<string, object?>> ReferralsAsync(int clientId)
    {
        return await Http.GetAsync(string.Format("/api/affiliates_adv/{clientId}/referrals", clientId));
    }

    public async Task<Dictionary<string, object?>> PayoutsAsync(int clientId)
    {
        return await Http.GetAsync(string.Format("/api/affiliates_adv/{clientId}/payouts", clientId));
    }

    public async Task<Dictionary<string, object?>> CampaignsAsync(int clientId)
    {
        return await Http.GetAsync(string.Format("/api/affiliates_adv/{clientId}/campaigns", clientId));
    }

    public async Task<Dictionary<string, object?>> AuditAsync(int clientId)
    {
        return await Http.GetAsync(string.Format("/api/affiliates_adv/{clientId}/audit", clientId));
    }

    public async Task<Dictionary<string, object?>> ActivateAsync(int clientId, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/affiliates_adv/{clientId}/activate", clientId), data);
    }

    public async Task<Dictionary<string, object?>> SetCommissionPlanAsync(int clientId, int commissionId)
    {
        return await Http.PostAsync(string.Format("/api/affiliates_adv/{clientId}/commission-plan/{commissionId}", clientId, commissionId), null);
    }

    public async Task<Dictionary<string, object?>> CreateVoucherAsync(int clientId, int planId, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/affiliates_adv/{clientId}/vouchers/{planId}", clientId, planId), data);
    }

    public async Task<Dictionary<string, object?>> DeleteVoucherAsync(int clientId, int voucherId)
    {
        return await Http.DeleteAsync(string.Format("/api/affiliates_adv/{clientId}/vouchers/{voucherId}", clientId, voucherId));
    }

    public async Task<Dictionary<string, object?>> SetLandingPageAsync(int clientId, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/affiliates_adv/{clientId}/landing-page", clientId), data);
    }
}
